Foundation/ISO8601DateFormatter.swift

```diff
@@ -79,8 +79,44 @@ open class ISO8601DateFormatter : Formatter, NSSecureCoding {
         super.init()
     }
     
-    public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
-    open override func encode(with aCoder: NSCoder) { NSUnimplemented() }
+    public required init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            fatalError("ISO8601DateFormatter cannot be decoded by non-keyed archivers")
+        }
+        
+        let options = Options(rawValue: UInt(aDecoder.decodeInteger(forKey: "NS.formatOptions")))
+        let timeZone: NSTimeZone?
+        
+        if aDecoder.containsValue(forKey: "NS.timeZone") {
+            if let tz = aDecoder.decodeObject(of: NSTimeZone.self, forKey: "NS.timeZone") {
+                timeZone = tz
```

Comment:
It may be clearer to keep all the time zone-related statements together. I would just make it

```
if let tz = aDecoder.decodeObject(of: NSTimeZone.self,
                                                           forKey: "NS.timeZone")?._swiftObject {
    self.timeZone = tz
}
```

Then you can do away with lines 88 (`let timeZone: NSTimeZone?`), 97-99, and 102-105 altogether.

Reply:
That's too many things to parse in a single line, I feel.

Reply:
Fair enough. But can they be grouped together a little more?